packages/perseus-editor/src/widgets/image-editor.tsx

```diff
@@ -288,25 +318,19 @@ const ImageEditor: any = createReactClass({
             });
             return;
         }
-        this.setState({backgroundImageError: ""});
 
-        // We update our background image prop after the image loads below. To
-        // avoid weirdness when we change to a very slow URL, then a much
-        // faster URL, we keep track of the URL we're trying to change to.
-        this._leadingUrl = url;
+        // Clear previous errors
+        this.setState({backgroundImageError: ""});
 
-        if (!url) {
-            this.setUrl(url, 0, 0, silent);
-            return;
+        try {
+            const size = await Util.getImageSizeModern(url);
+            this.setUrl(url, size[0], size[1], true);
+        } catch (error) {
+            this.setState({
+                backgroundImageError:
+                    "There was an error loading the image URL",
```

Comment:
I wonder if we want to provide the `error` text in this message?  For learner-facing I think we'd want more polish, but the source error might give more insight to the content author why the image couldn't load.  What do you think?